azure-quantum/azure/quantum/target/ionq.py

```diff
@@ -82,3 +112,68 @@ def submit(
             input_params=input_params,
             **kwargs
         )
+
+    def calculate_cost(
+        self,
+        circuit: Dict[str, Any],
+        num_shots: int,
+        cost_1q: float=0.00003,
+        cost_2q: float=0.0003,
+        min_cost: float=1.0
+    ) -> float:
+        """Calculate the cost of submittng a circuit to IonQ targets.
+        Optionally, you can provide the number of gate and measurement operations
+        manually.
+        The actual cost charged by the provider may differ from this calculation.
+        
+        Specify pricing details for your area to get most accurate results.
+        By default, this function charges cost_1q=0.00003 USD for a single-qubit gate,
+        cost_2q=0.0003 USD for a two-qubit gate with a total minimum cost of $1.-
+        per circuit.
+
+        For the most current pricing details, see
+        https://docs.microsoft.com/en-us/azure/quantum/provider-ionq#pricing
+        Or find your workspace and view pricing options in the "Provider" tab
+        of your workspace: http://aka.ms/aq/myworkspaces
+
+        :param circuit: Quantum circuit in IonQ JSON format (for examples,
+            see: https://docs.ionq.com/#section/Sample-JSON-Circuits)
+        :type circuit: Dict[str, Any]
+        :param num_shots: Number of shots, defaults to None
+        :type num_shots: int
+        :param cost_1q: The cost of running a single-qubit gate
+            for one shot, defaults to 0.00003
+        :type cost_1q: float, optional
+        :param cost_2q: The cost of running a double-qubit gate
+            for one shot, defaults to 0.0003
+        :type cost_2q: float, optional
+        :param min_cost: The minimum cost for running a job, defaults to 1.0
+        :type min_cost: float, optional
+        """
+        if self.name == "ionq.simulator": 
+            return 0.0
+
+        def is_1q_gate(gate: Dict[str, Any]):
+            return gate.get("gate") in self.GATES_1Q and (
```

Comment:
can we check instead how many target qubits are there in the gate?

Reply:
Currently, we're checking both. This list is exhaustive and taken from the IonQ docs: https://docs.ionq.com/#section/Sample-JSON-Circuits